### PR TITLE
Fix condition for using uv in some jobs

### PR DESCRIPTION
**Pull request by potiuk**
The condition to check if pip is forced in the CI has a bug (GitHub actions bool behaviour is somewhat problematic) and uv has not been used really for some of the jobs. This is now fixed.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by romsharon98 — APPROVED** [review]

**Comment by gopidesupavan** [comment]
Yeah this somewhat problematic thing difficult to debug :) 

I looked at their documentation, they some suggestion to use `fromjson`  

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#fromjson

**Review by gopidesupavan — APPROVED** [review]

**Review by hussein-awala — APPROVED** [review]

File: .github/workflows/build-images.yml [hunk 1]
```diff
@@ -204,7 +204,7 @@ jobs:
       pull-request-target: "true"
       is-committer-build: ${{ needs.build-info.outputs.is-committer-build }}
       push-image: "true"
-      use-uv: ${{ needs.build-info.outputs.force-pip && 'false' || 'true' }}
+      use-uv: ${{ needs.build-info.outputs.force-pip=='true' && 'false' || 'true' }}
```
Comment (Lee-W) [inline .github/workflows/build-images.yml]:
```suggestion
      use-uv: ${{ needs.build-info.outputs.force-pip == 'true' && 'false' || 'true' }}
```

nits: spaces seem to be used elsewhere 👀

File: .github/workflows/build-images.yml [hunk 2]
```diff
@@ -249,7 +249,7 @@ jobs:
...
```
Comment (Lee-W) [inline .github/workflows/build-images.yml]:
```suggestion
      use-uv: ${{ needs.build-info.outputs.force-pip == 'true' && 'false' || 'true' }}
```

File: .github/workflows/finalize-tests.yml [hunk 3]
```diff
@@ -149,7 +149,7 @@ jobs:
       python-versions: ${{ inputs.python-versions }}
       branch: ${{ inputs.branch }}
       constraints-branch: ${{ inputs.constraints-branch }}
-      use-uv: ${{ needs.build-info.outputs.force-pip && 'false' || 'true' }}
+      use-uv: ${{ needs.build-info.outputs.force-pip=='true' && 'false' || 'true' }}
```
Comment (Lee-W) [inline .github/workflows/finalize-tests.yml]:
```suggestion
      use-uv: ${{ needs.build-info.outputs.force-pip == 'true' && 'false' || 'true' }}
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -642,7 +642,7 @@ jobs:
       kubernetes-versions-list-as-string: ${{ needs.build-info.outputs.kubernetes-versions-list-as-string }}
       kubernetes-combos-list-as-string: ${{ needs.build-info.outputs.kubernetes-combos-list-as-string }}
       include-success-outputs: ${{ needs.build-info.outputs.include-success-outputs }}
-      use-uv: ${{ needs.build-info.outputs.force-pip && 'false' || 'true' }}
+      use-uv: ${{ needs.build-info.outputs.force-pip=='true' && 'false' || 'true' }}
```
Comment (Lee-W) [inline .github/workflows/ci.yml]:
```suggestion
      use-uv: ${{ needs.build-info.outputs.force-pip == 'true' && 'false' || 'true' }}
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -559,7 +559,7 @@ jobs:
       default-python-version: ${{ needs.build-info.outputs.default-python-version }}
       branch: ${{ needs.build-info.outputs.default-branch }}
       push-image: "true"
-      use-uv: ${{ needs.build-info.outputs.force-pip && 'false' || 'true' }}
+      use-uv: ${{ needs.build-info.outputs.force-pip=='true' && 'false' || 'true' }}
```
Comment (Lee-W) [inline .github/workflows/ci.yml]:
```suggestion
      use-uv: ${{ needs.build-info.outputs.force-pip == 'true' && 'false' || 'true' }}
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -206,7 +206,7 @@ jobs:
       platform: "linux/amd64"
       python-versions: ${{ needs.build-info.outputs.python-versions }}
       branch: ${{ needs.build-info.outputs.default-branch }}
-      use-uv: ${{ needs.build-info.outputs.force-pip && 'false' || 'true' }}
+      use-uv: ${{ needs.build-info.outputs.force-pip=='true' && 'false' || 'true' }}
```
Comment (Lee-W) [inline .github/workflows/ci.yml]:
```suggestion
      use-uv: ${{ needs.build-info.outputs.force-pip == 'true' && 'false' || 'true' }}
```

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -272,7 +272,7 @@ jobs:
       latest-versions-only: ${{ needs.build-info.outputs.latest-versions-only }}
       include-success-outputs: ${{ needs.build-info.outputs.include-success-outputs }}
       debug-resources: ${{ needs.build-info.outputs.debug-resources }}
-      use-uv: ${{ needs.build-info.outputs.force-pip && 'false' || 'true' }}
+      use-uv: ${{ needs.build-info.outputs.force-pip=='true' && 'false' || 'true' }}
```
Comment (Lee-W) [inline .github/workflows/ci.yml]:
```suggestion
      use-uv: ${{ needs.build-info.outputs.force-pip == 'true' && 'false' || 'true' }}
```

**Pull request merged**